### PR TITLE
need to fix upload issues (work in progress)

### PR DESCRIPTION
**Pull request by solderq35**
The download and user login issues are fixed on the production site, but the upload path is now broken.

I ported over some changes from the download path fix (e.g. adding `resolve` for delete and upload paths), although I wasn't able to get it working locally. I think I might need some valid location data or remove the location data function for it to work (more info below).

Relevant screenshots:
Console Errors

json

Note in the second screenshot how the location data field is empty; I have a hunch this may be an underlying issue.

I changed `http://extreme-ip-lookup.com/json/` in `src/utils/api/user.js` to use the https protocol instead (`https://extreme-ip-lookup.com/json/`), but I think this alone will not fix it, as it seems like extreme-ip-lookup requires an API key (I'm not sure if this was changed since 2020 or whatever). Without an  API key some information is missing such as city info.

Extreme-API key

Note we could just use the example key I think (ex: https://extreme-ip-lookup.com/json/?key=demo2), or sign up for our own key.

Alternatively, it's arguable we don't need the user location to begin with, we could try to edit the location functions out of the code if it messes up the upload request.

TL;DR: Need input on the location data stuff, assuming it would fix the upload path anyways. Also I think we might have to redeploy the backend to test this, which may mean we have to deploy to production to test.

### CODE REVIEW

**Comment by solderq35** [comment]
EDIT: Porting to API key / https protocol on the extreme-ip-lookup request fixed the "mixed content" error, but it still fails for some reason, as does the upload path.

test build update

Tested on the test build site (http://carbon-calculator.s3-website-us-west-2.amazonaws.com/#/). The backend changes were not deployed here, so this may not be 100% accurate either.

**Comment by solderq35** [comment]
I don't know if it proves anything but I was able to upload some data via Curl using some tweaking.

Using syntax from https://www.baeldung.com/curl-rest

`curl -d <data> -H 'Content-Type: application/json' http://localhost:3000/upload <append cookie data from download cookie>`

Example (this is Windows syntax): 
```
curl -d "{\"date\":\"12/29/2022\",\"location\":{},\"totals\":[0,0,4498.6,0,170.2]}" -H 'Content-Type: application/json' http://localhost:3000/upload ^
  -H "authority: api.sustainability.oregonstate.edu" ^
  -H "accept: application/json, text/plain, */*" ^
  -H "accept-language: en-US,en;q=0.9" ^
  -H "cookie: redirect=http^%^3A^%^2F^%^2Flocalhost...; 
  ``` 
  (rest of cookie redacted)
  
  Screenshot of successful upload (local API):
  upload
  
  I think this should show that the location data (or lack thereof) is not an issue I think.
  
  - I'm not sure if the upload request gets the right authorization cookie (as I had to get it from the download request)
  - I'm not sure if there is still some CORS issue such as wrong `Access-Control-Allow-Origin`

**Comment by solderq35** [comment]
I was able to fix the getlocation function at least on the test build, but not the upload path itself. The upload request is giving a seemingly wrong access-control-allow-origin URL (should be myco2.sustainability.oregonstate.edu), but it is listed as api.sustainability.oregonstate.edu.

wrong origin

**Comment by solderq35** [comment]
Decided to remove getlocation function as it's not displayed to the user or otherwise used anyways.

**Comment by solderq35** [comment]
I tested out some configurations on the `rework-login` branch (that is currently setup for some local testing): https://github.com/OSU-Sustainability-Office/osu_carbon_calculator_update_project/tree/rework-user-login

network

template

As you can see, setting the `AllowOrigin` value in `backend/template.yaml` to be `http://localhost:8080` (the frontend URL for local testing) ensured that the access-control-allow-origin from response header (`http://localhost:8080`) matched the origin from the request header (`http://localhost:8080`). 

By this logic I think that setting `AllowOrigin` in `backend/template.yaml` to `https://myco2.sustainability.oregonstate.edu` should fix the upload path in production, so that's a change I've made for this PR.
